test/curve_fit.jl

```diff
@@ -14,4 +14,18 @@ let
     # can also get error estimates on the fit parameters
     errors = estimate_errors(fit)
     @assert norm(errors - [0.017, 0.075]) < 0.01
+
+    # some example data
+    yvars = 1e-4*randn(length(xdata))
+    ydata = model(xdata, [1.0, 2.0]) + sqrt(yvars).*randn(length(xdata))
+
+    fit = curve_fit(model, xdata, ydata, 1./sqrt(yvars), [0.5, 0.5])
+    println("norm(fit.param - [1.0, 2.0]) < 0.05 ? ", norm(fit.param - [1.0, 2.0]))
+    @assert norm(fit.param - [1.0, 2.0]) < 0.05
+    @test fit.converged
+
+    # can also get error estimates on the fit parameters
+    errors = estimate_errors(fit,yvars)
```

Comment:
does it feel right to you that the interface for `estimate_errors` changes when the data points are not weighted evenly? the alternative would be to store enough information in `LsqFitResult`.